### ISSUE TITLE
topsStack bug: occasionally  relativeOrbitNumber to track number conversion wrong

### ISSUE THREAD

**Issue by falkamelung** [isce-framework/isce2#104]
Occasionally, in topsStack, the relativeOrbitNumber is not properly converted to track number in the /master/IW1.xml file (in about 10-20% of cases). Here an example for  descending track 87 covering Kilauea volcano.   

Here the *.SAFE file for 2018-05-05 showing the correct relativeOrbitNumber of 87:
``` 
grep OrbitNumber S1B_IW_SLC__1SDV_20180505T161525_20180505T161552_010788_013B7D_25D2.SAFE/*
S1B_IW_SLC__1SDV_20180505T161525_20180505T161552_010788_013B7D_25D2.SAFE/manifest.safe:            <safe:relativeOrbitNumber type="start">87</safe:relativeOrbitNumber>
```

However, the master/IW1.xml file shows a track number of 41:
```
/scratch/05861/tg851601/KilaueaSenDT87[1104] head -1422 master/IW1.xml | tail -20
                <property name="sensingstop">
                    <value>2018-05-05 16:15:32.858289</value>
                    <doc>UTC time corresponding to last line of burst SLC</doc>
                </property>
                <property name="startingrange">
                    <value>799103.829224447</value>
                    <doc>Slant range to first pixel in m</doc>
                </property>
                <property name="swathnumber">
                    <value>1</value>
                    <doc>Swath number for bookkeeping</doc>
                </property>
                <property name="terrainheight">
                    <value>0.0</value>
                    <doc>Average terrain height used for focusing</doc>
                </property>
                <property name="tracknumber">
                    <value>41</value>
                    <doc>Track number for bookkeeping</doc>
                </property>
```

Weirdly, some slave granules don't show  the correct  track number, whereas others do (the first one shows 41 and the second correctly 87)
```
//login4/scratch/05861/tg851601/KilaueaSenDT87[1056] head -1422  slaves/20180517/IW1.xml | tail -20
                <property name="sensingstop">
                    <value>2018-05-17 16:15:33.369091</value>
                    <doc>UTC time corresponding to last line of burst SLC</doc>
                </property>
                <property name="startingrange">
                    <value>799103.829224447</value>
                    <doc>Slant range to first pixel in m</doc>
                </property>
                <property name="swathnumber">
                    <value>1</value>
                    <doc>Swath number for bookkeeping</doc>
                </property>
                <property name="terrainheight">
                    <value>0.0</value>
                    <doc>Average terrain height used for focusing</doc>
                </property>
                <property name="tracknumber">
                    <value>41</value>
                    <doc>Track number for bookkeeping</doc>
                </property>
//login4/scratch/05861/tg851601/KilaueaSenDT87[1057] head -1422  slaves/20190331/IW1.xml | tail -20
                <property name="sensingstop">
                    <value>2019-03-31 16:16:19.877707</value>
                    <doc>UTC time corresponding to last line of burst SLC</doc>
                </property>
                <property name="startingrange">
                    <value>799103.9365244998</value>
                    <doc>Slant range to first pixel in m</doc>
                </property>
                <property name="swathnumber">
                    <value>1</value>
                    <doc>Swath number for bookkeeping</doc>
                </property>
                <property name="terrainheight">
                    <value>0.0</value>
                    <doc>Average terrain height used for focusing</doc>
                </property>
                <property name="tracknumber">
                    <value>87</value>
                    <doc>Track number for bookkeeping</doc>
                </property>
```

If you want to download example data using ssara:
```
ssara_federated_query-cj.py --platform=SENTINEL-1A,SENTINEL-1B --relativeOrbit=87 --intersectsWith='Polygon((-155.80 19.10, -155.80 19.90, -154.50 19.90, -154.50 19.10, -155.80 19.10))' -s=2018-05-05 --parallel=50 --print --download
```

The topsStack workflow is initiated using:
```
stackSentinel.py --slc_directory /scratch/05861/tg851601/KilaueaSenDT87/SLC --orbit_directory /work/05861/tg851601/stampede2/insarlab/S1orbits --aux_directory /work/05861/tg851601/stampede2/insarlab/S1aux --working_directory /scratch/05861/tg851601/KilaueaSenDT87 --dem DEM/dem.dem.wgs84 --num_connections 5 --num_overlap_connections 3 --swath_num '1 2' --bbox '19.1 19.9 -155.8 -154.5' --azimuth_looks 7 --range_looks 19 --filter_strength 0.2 --esd_coherence_threshold 0.85 --snr_misreg_threshold 10 --unw_method snaphu --polarization vv --coregistration NESD --workflow interferogram
```

**Comment by piyushrpt** [comment]
I believe some of the very early S1 products did not have trackNumber and we had to compute it from absolute orbit number. 

https://github.com/isce-framework/isce2/blob/e0b7ad05a1c581aadc2296dc2dd016dba2e273dd/components/isceobj/Sensor/TOPS/Sentinel1.py#L601

If the logic here can be extended to account for S1B (probably equation needs an additional constant to be added before modulo) - this should be fixed.

**Comment by falkamelung** [comment]
Ahh, right. It happens only for S1B. It would be
`(orbitnumber-27)%175 + 1`

**Comment by piyushrpt** [comment]
Closed by #109
